### DISCUSSION TITLE
How to use a custom directory to store files within the AdonisJS 5 API

### DISCUSSION THREAD

**Discussion by LeadcodeDev** [adonisjs/core#1567] [Help] [answered]
Hello !
I followed the documentation (https://preview.adonisjs.com/guides/http/file-uploads/). Following Harminder Virk's message in discord, I manage to upload files to a custom directory.

```ts
public async store({ request }: HttpContextContract) {
	const timestamp = new Date().getTime()
	const serviceId = request.input('service_id')
	const document = request.file('picture', {
		extnames: ['jpg', 'jpeg', 'png', 'svg']
	})

	if (document?.hasErrors) return document.errors
	const picture = await Picture.create({
		filename: `${timestamp}.${document!.extname}`,
		serviceId
	})

	await document?.move(Application.makePath(`../public/storage/services`), {
		name: `${timestamp}.${document.extname}`
	})
	await Picture.query().where('id', picture.id).update({
		filename: document?.fileName,
		label: document!.clientName
	})

	return { picture }
}
```
My uplaod folder is work !

My problem is that if I follow the documentation 100% my files are uploads in the public folder of my application (so that I can retrieve my images on my frontend) but when I have to re-build my application my uploads are deleted because the public folder is purged and then re-created. With my controller above, my uploads are saved in a public folder at the root of my application.

The problem is that with this method, I can't get my image path for my nuxtJS front.

Can you help me ?

**Comment by thetutlage** [comment, marked as answer]
You are simply relying on the `/public` folder to serve images for you. As shared earlier in the discord, you should just write a 2 liner route to serve images from a custom directory.

- I will store the files inside `../storage/uploads`.
- Then write the following route to server uploaded files.
   ```ts
   Route.get('uploads/:filename', async ({ response, params }) => {
     response.download(Application.makePath('../storage/uploads', params.filename))
   })
   ```
- Finally access files by visiting `/uploads/<filename>`

**Reply by LeadcodeDev** [reply]
You saved me 😍 Everything works nickel ! Thank you very much !

**Reply by ericmp33** [reply]
Here is the `Aplication` import:
```js
import Application from '@ioc:Adonis/Core/Application'
```
